src/platform/Linux/bluez/ChipDeviceScanner.cpp

```diff
@@ -179,9 +198,15 @@ CHIP_ERROR ChipDeviceScanner::StopScan()
     if (PlatformMgrImpl().GLibMatterContextInvokeSync(MainLoopStopScan, this) != CHIP_NO_ERROR)
     {
         ChipLogError(Ble, "Failed to schedule BLE scan stop.");
+        mScannerLock.unlock();
         return CHIP_ERROR_INTERNAL;
     }
 
+    // Stop scanning and return to initialization state
+    mCurrentState = ChipDeviceScannerState::SCANNER_INITIALIZED;
```

Comment:
Don't we need some kind of `STOPPING` state given that the stop occurs asynchronously? Though it's probably not an issue as the start is also executed in the GLIB context...

Reply:
Here `mScannerLock` is to avoid race conditions when multiple threads call `StopScan`, so `STOPPING` is not needed.